### PR TITLE
Added typings for tough-cookie-filestore

### PR DESCRIPTION
**Pull request by friedow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Adding a new definition:
- [x] The package does not already provide its own types
- [x] If this is for an NPM package, match the name.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/tough-cookie-filestore/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+      "module": "commonjs",
+      "lib": [
+          "es6"
+      ],
+      "noImplicitAny": true,
+      "noImplicitThis": true,
+      "strictNullChecks": false,
```
Comment (plantain-00) [inline types/tough-cookie-filestore/tsconfig.json]: For new packages, it should be true here

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by friedow** [comment]
Thanks for the fast review @plantain-00. I incorporated your changes as requested.

**Review by plantain-00 — APPROVED** [review]

File: types/tough-cookie-filestore/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// Definitions by: Christian Friedow <https://github.com/friedow>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Cookie {
+    domain: string;
+    path: string;
+    key: string;
+}
+
+export default class FileCookieStore {
```
Comment (uniqueiniquity) [inline types/tough-cookie-filestore/index.d.ts]: It seems like the [underlying library](https://unpkg.com/tough-cookie-filestore@0.0.1/index.js) is not using `export default`, so this .d.ts file should not either. Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html). Note that if you want to use `import name from 'package';` in your consuming code, you can use the `esModuleInterop` compiler setting.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/tough-cookie-filestore/index.d.ts (hunk 2 above)
Comment (friedow) [reply types/tough-cookie-filestore/index.d.ts]: What exactly would you change here? I mean it still remains a class that gets exported, right?

Comment (uniqueiniquity) [reply types/tough-cookie-filestore/index.d.ts]: As discussed at that handbook link, I'd do:
```ts
export class FileCookieStore {
// define your class
}

export = FileCookieStore
```
What you have now is a class that gets exported on the `.default` property of the resulting module, which is fine when users of this package have `esModuleInterop` enabled, but if they don't, then it'll cause a runtime error.

**Comment by friedow** [comment]
After changing the files according to the handbook, I reopened this PR at #28596. @uniqueiniquity, @plantain-00 would you please have a look?

**Pull request closed without merging**